### DISCUSSION TITLE
Optimisation des options de sécurité (Captcha et 2FA)

### DISCUSSION THREAD

**Discussion by JGG-Apps** [JeromeDevome/GRR#361] [Idées]
**Votre demande de fonctionnalité est-elle liée à un problème ?

Je ne pourrais pas dire un problème, mais un enjeu sur la pérennité d'utilisation du produit. En raison de la nature du Web en général, afin d'avoir un produit répondant à un niveau de protection adéquat contre les attaques, il y a actuellement une grande lacune dans la solution GRR.

**Décrivez la solution que vous souhaitez**

1. Dans toutes les sections où il y a possibilité de s'authentifier, une mécanique 2FA (TOTP ou autre) devrait être mis en place.

2. Dans toutes les sections où il est possible de consulter des données ou en insérer sans authentification (invité), le Captcha v2 devrait être  mis à en place.

3. Dans toutes les sections où une demande de changement important (administratif) doit avoir lieu, le Captcha v2 devrait être mis  en place.

4. Dans l'aspect "esthétique" et tout en même temps pour permettre d'intégrer un Captcha avec l'accès invité, il serait intéressant (chose que j'aimerais depuis longtemps) qu'une page d'accueil soit possible plutôt que l'agenda soit directement affiché comme page principale. Un genre de portail de bienvenue dans lequel on pourrait alimenter logo, quelques informations de base, but, objectif de l'agenda déployé. En fait une page qui serait similaire à la page "login" lorsque l'authentification est requise... même si l'authentification n'est pas nécessaire et dans laquelle on pourrait sélectionner "Accéder en tant qu'invité" et si ce soit sans Captcha n'est pas fait, l'accès au reste du site serait impossible.

**Décrivez les alternatives que vous avez envisagées**
Ma demande contient plusieurs éléments, j'espère qu'elles seront tous réalisées, mais au moins débuter par l'une d'elles :)

**Contexte supplémentaire**
Merci de faire évoluer le produit !

**Comment by JeromeDevome** [comment]
Bonjour,
Vous décrivez un système idéal. On pourrait même rajouter :
- Obligatoin de complexité de mot de passe
- Verrouillage si trop d'essaie dedans la saisie de mot de passe
- Blocage d'ip , trop de tentatives
- Expiration de mot de passe

Tout cela demande du temps au développement, si vous pouvez développer les fonctions, nous vous en remercions.

**Comment by JeromeDevome** [comment]
Bonjour,
Nous sommes pas encore au MFA, mais nous nous avons développé pour la prochaine version, la complexité de mot de passe (5134ddb8).
